src/test/java/formflow/library/submissions/actions/CalculateTotalBeforeSave.java

```diff
@@ -0,0 +1,21 @@
+package formflow.library.submissions.actions;
+
+import formflow.library.config.submission.Action;
+import formflow.library.data.Submission;
+import java.util.List;
+import java.util.Map;
+
+@SuppressWarnings("unused")
+public class CalculateTotalBeforeSave implements Action {
+
+  public void run(Submission submission, String _id) {
+    List<Map<String, Object>> subflow = (List<Map<String, Object>>) submission.getInputData()
+        .get("income");
+    var totalIncome = subflow.stream()
+        .map(e -> Integer.parseInt((String) e.get("textInput")))
+        .reduce(Integer::sum)
+        .get();
```

Comment:
We do allow folks to enter income having a decimal value (example: $10.50), which would be lost here with an Integer. This should probably be a Float/Double instead.   Or we need away to allow for both scenarios, where truncating the fractional part of the value is okay?

How much do you care about it being an Integer versus Double?    I know the fractional value doesn't amount to much when we are talking about hundreds or thousands of dollars - we just need to align on this.

Reply:
Hm, I just used `int` to have something really basic for testing. Ive seen it both ways for real income inputs so no strong preference here.

Reply:
Yeah, no strong preference here either. Since we've assumed a fractional value so far, would you mind changing this to a Double?